e2e/runner/v2_setup_zeta.go

```diff
@@ -67,3 +74,37 @@ func (r *E2ERunner) SetZEVMContractsV2() {
 	ensureTxReceipt(txProxy, "Gateway proxy deployment failed")
 	ensureTxReceipt(txTestDAppV2, "TestDAppV2 deployment failed")
 }
+
+// UpdateChainParamsERC20CustodyContract update the erc20 custody contract in the chain params
+// this operation is used when transitioning to new smart contract architecture where a new ERC20 custody contract is deployed
+func (r *E2ERunner) UpdateChainParamsERC20CustodyContract() {
+	res, err := r.ObserverClient.GetChainParams(r.Ctx, &observertypes.QueryGetChainParamsRequest{})
+	require.NoError(r, err)
+
+	evmChainID, err := r.EVMClient.ChainID(r.Ctx)
+	require.NoError(r, err)
+
+	// find old chain params
+	var (
+		chainParams *observertypes.ChainParams
+		found       bool
+	)
+	for _, cp := range res.ChainParams.ChainParams {
+		if cp.ChainId == evmChainID.Int64() {
+			chainParams = cp
+			found = true
+			break
+		}
+	}
+	require.True(r, found, "Chain params not found for chain id %d", evmChainID)
+
+	// update with the new ERC20 custody contract address
+	chainParams.Erc20CustodyContractAddress = r.ERC20CustodyV2Addr.Hex()
```

Comment:
nit: I'd consider reviewing the whole codebase for naming discrepancies like this:
```suggestion
	chainParams.ERC20CustodyContractAddress = r.ERC20CustodyV2Addr.Hex()
```

Reply:
I agree with those. Since we're renaming a field here, might introduce extensive changes
Created an issue for it: https://github.com/zeta-chain/node/issues/2714